src/symbol.cc

```diff
@@ -113,18 +113,34 @@ std::vector<std::string> ELF::NeededLibs() {
   return needed;
 }
 
-unsigned long ELF::GetThreadState() {
+unsigned long ELF::GetThreadState(int *version) {
+  int ver = 0;
+  unsigned long addr = 0;
   const shdr_t *s = shdr(dynsym_);
   const shdr_t *d = shdr(dynstr_);
   for (uint16_t i = 0; i < s->sh_size / s->sh_entsize; i++) {
     const sym_t *sym =
         reinterpret_cast<const sym_t *>(p() + s->sh_offset + i * s->sh_entsize);
     const char *name =
         reinterpret_cast<const char *>(p() + d->sh_offset + sym->st_name);
-    if (strcmp(name, "_PyThreadState_Current") == 0) {
-      return static_cast<unsigned long>(sym->st_value);
+    if (!addr && strcmp(name, "_PyThreadState_Current") == 0) {
+      addr = static_cast<unsigned long>(sym->st_value);
+    } else if (!ver) {
+      if (strcmp(name, "PyString_Type") == 0) {
+        // if we find PyString_Type, it's python 2
+        ver = 2;
+      } else if (strcmp(name, "PyBytes_Type") == 0) {
+        // if we find PyBytes_Type, it's python 3
+        ver = 3;
+      }
     }
+    if (ver && addr) {
+      break;
+    }
+  }
+  if (version != nullptr) {
```

Comment:
I actually removed this check in the most recent version, since the function is only called in one place w/o a null pointer.